rootex/core/audio/audio_source.h

```diff
@@ -4,24 +4,36 @@ class StreamingAudioBuffer;
 class StaticAudioBuffer;
 
 typedef unsigned int ALuint;
+typedef float ALfloat; // doubt(32-bit IEEE 754 floating-point)
 
 /// Convert minutes to seconds
 #define MIN_TO_S 60.0f
 
 /// An interface for an audio source in the game world.
 class AudioSource
 {
+
 protected:
 	ALuint m_SourceID;
 
 	/// RTTI for storing if the audio buffer is being streamed
-	bool m_IsStreaming; 
+	bool m_IsStreaming;
 	AudioSource(bool isStreaming);
 	virtual ~AudioSource();
 
 public:
+
+	enum class AttenuationModel
+	{
+		linear,
```

Comment:
Change to PascalCase. 

Also since this maps directly to AL_LINEAR_DISTANCE, do:
enum class AttenuationModel {Linear = AL_DISTANCE_LINEAR}
Same with the others. Then you will not need to switch case over the attenuation model, you will be able to send the exact same value directly to OpenAL.